src/server/bin/plugin/npmInstall.js

```diff
@@ -0,0 +1,37 @@
+﻿var Promise = require('bluebird');
+var fs = require('fs');
+var path = require('path');
+var exec = require('child_process').exec;
+
+module.exports = function (dest, logger) {
+  return new Promise(function (resolve, reject) {
+    //throw an exception if package.json does not exist
+    try {
+      var packageFile = path.join(dest, 'package.json');
+      fs.statSync(packageFile);
+    } catch (e) {
+      if (e.code !== 'ENOENT')
+        throw e;
+
+      return reject(new Error('Plugin does not contain package.json file'));
+    }
+
+    var cmd = (process.env.NPM) ? process.env.NPM : 'npm';
+    cmd += ' install';
+
+    var child = exec(cmd, { cwd: dest });
```

Comment:
Modified the --silent option so that it applies to both log and error messages per a conversation with @rashidkpc.